### PR TITLE
Ac 1235 react template

### PR DESCRIPTION
**Pull request by Luca-Ferrera**
Hooks removed to avoid warning during testing.
`addLocaleData` removed since no longer used in V3 of react-intl.
See [documentation](https://github.com/formatjs/react-intl/blob/master/docs/Upgrade-Guide.md#migrate-to-using-native-intl-apis )

### CODE REVIEW

File: src/components/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/*
+* Insert in this folder your react component.
```
Comment (dibericky) [inline src/components/index.js:2]: I think we could delete this file

File: src/components/utils/PromiseComponent/__tests__/index.test.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react'
+import {mount} from 'enzyme'
+
+import PromiseComponent from '..'
+import {FormattedMessage, IntlProvider} from 'react-intl'
```
Comment (dibericky) [inline src/components/utils/PromiseComponent/__tests__/index.test.js]: move this line above with others module dependencies

File: src/containers/index.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+/*
+*  Insert in this folder your react container
```
Comment (dibericky) [inline src/containers/index.js:2]: delete this file

File: README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
```
Comment (davidebianchi) [inline README.md:3]: We should add how to create docker image docs

File: package.json [hunk 5]
```diff
@@ -0,0 +1,76 @@
+{
+  "name": "afaf",
```
Comment (davidebianchi) [inline package.json]: afaf?

File: package.json [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "npm-run-all lint coverage"
+    }
+  },
+  "eslintConfig": {
+    "extends": "react-app"
+  },
+  "browserslist": [
```
Comment (davidebianchi) [inline package.json]: browserlist should not be different? I think that this is the old configuration

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,26 @@
+FROM nexus.mia-platform.eu/core/static-files:3.2.2 as build
+
+USER root
```
Comment (fredmaggiowski) [inline Dockerfile]: Let's not use root user

File: README.md (hunk 4 above)
Comment (fredmaggiowski) [inline README.md:3]: The file is missing the title with the service name

File: src/App.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import './App.css'
+
+class App extends Component {
+  render () {
+    return (
+      <div className='App'>
+        <header className='App-header'>
+          <img alt='logo' className='App-logo' src={logo} />
+          <p>
+            <FormattedMessage id={'edit'} />
```
Comment (fredmaggiowski) [inline src/App.js:30]: We can use a single formatted message and passing the filename as values prop

File: src/components/index.js (hunk 1 above)
Comment (fredmaggiowski) [reply src/components/index.js:2]: We can leave the file or use a `.gitkeep` instead; i've no strong opinion about this... Maybe the `.gitkeep` has less impacts on how the user will handle its files.

Comment (fredmaggiowski) [reply src/components/index.js:2]: note: this applies to all empty index files in the sibling directories.

File: src/components/utils/PromiseComponent/__tests__/index.test.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+ */
+
+import React from 'react'
+import {mount} from 'enzyme'
+import {FormattedMessage, IntlProvider} from 'react-intl'
+
+import PromiseComponent from '..'
+
+const italian = {'test': 'Questa è una stringa in italiano'}
+
```
Comment (fredmaggiowski) [inline src/components/utils/PromiseComponent/__tests__/index.test.js:24]:
```suggestion
```

File: src/components/utils/PromiseComponent/__tests__/index.test.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+      <PromiseComponent promiseFunction={promiseFunction}>
+        {data => children(data)}
+      </PromiseComponent>
+    )
+    setImmediate(() => {
+      expect(element.find('div').text()).toEqual('Error')
+      expect(children).toHaveBeenCalledTimes(0)
+    })
+  })
+  it('renders div with Loading... when usePromise returns data undefined and isError false', () => {
```
Comment (fredmaggiowski) [inline src/components/utils/PromiseComponent/__tests__/index.test.js]:
```suggestion

  it('renders div with loading message when usePromise returns data undefined and isError false', () => {
```

**Review by fredmaggiowski — CHANGES_REQUESTED** [review]

**Review by fredmaggiowski — APPROVED** [review]

File: src/App.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
+import React, {Component} from 'react'
```
Comment (davidebianchi) [inline src/App.js:17]: license on this file?

File: src/index.css [hunk 12]
```diff
@@ -0,0 +1,14 @@
+body {
```
Comment (davidebianchi) [inline src/index.css:17]: license

File: src/logo.svg [hunk 13]
```diff
@@ -0,0 +1,7 @@
...
...1 3.2-26.9 5.9-41.2 8 4.9-7.7 9.8-15.6 14.4-23.7 4.6-8 8.9-16.1 13-24.1zM421.2 430c-9.3-9.6-18.6-20.3-27.8-32 9 .4 18.2.7 27.5.7 9.4 0 18.7-.2 27.8-.7-9 11.7-18.3 22.4-27.5 32zm-74.4-58.9c-14.2-2.1-27.9-4.7-41-7.9 3.7-12.9 8.3-26.2 13.5-39.5 4.1 8 8.4 16 13.1 24 4.7 8 9.5 15.8 14.4 23.4zM420.7 163c9.3 9.6 18.6 20.3 27.8 32-9-.4-18.2-.7-27.5-.7-9.4 0-18.7.2-27.8.7 9-11.7 18.3-22.4 27.5-32zm-74 58.9c-4.9 7.7-9.8 15.6-14.4 23.7-4.6 8-8.9 16-13 24-5.4-13.4-10-26.8-13.8-39.8 13.1-3.1 26.9-5.8 41.2-7.9zm-90.5 125.2c-35.4-15.1-58.3-34.9-58.3-50.6 0-15.7 22.9-35.6 58.3-50.6 8.6-3.7 18-7 27.7-10.1 5.7 19.6 13.2 40 22.5 60.9-9.2 20.8-16.6 41.1-22.2 60.6-9.9-3.1-19.3-6.5-28-10.2zM310 490c-13.6-7.8-19.5-37.5-14.9-75.7 1.1-9.4 2.9-19.3 5.1-29.4 19.6 4.8 41 8.5 63.5 10.9 13.5 18.5 27.5 35.3 41.6 50-32.6 30.3-63.2 46.9-84 46.9-4.5-.1-8.3-1-11.3-2.7zm237.2-76.2c4.7 38.2-1.1 67.9-14.6 75.8-3 1.8-6.9 2.6-11.5 2.6-20.7 0-51.4-16.5-84-46.6 14-14.7 28-31.4 41.3-49.9 22.6-2.4 44-6.1 63.6-11 2.3 10.1 4.1 19.8 5.2 29.1zm38.5-66.7c-8.6 3.7-18 7-27.7 10.1-5.7-19.6-13.2-40-22.5-60.9 9.2-20.8 16.6-41.1 22.2-60.6 9.9 3.1 19.3 6.5 28.1 10.2 35.4 15.1 58.3 34.9 58.3 50.6-.1 15.7-23 35.6-58.4 50.6zM320.8 78.4z"/>
```
Comment (davidebianchi) [inline src/logo.svg:3]: What is this logo?

File: src/redux-store.js [hunk 14]
```diff
@@ -0,0 +1,52 @@
+import {createStore, combineReducers, applyMiddleware} from 'redux'
```
Comment (davidebianchi) [inline src/redux-store.js:17]: license

File: src/redux-store.js [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+  const composeEnhancers = composeWithDevTools({})
+
+  const store = createStore(
+    persistedReducer,
+    composeEnhancers(
+      applyMiddleware(...middlewares)
+    )
+  )
+
+  if (process.env.NODE_ENV === 'development') {
```
Comment (davidebianchi) [inline src/redux-store.js]: We could use NODE_ENV from config file

File: src/utilsTests.js [hunk 16]
```diff
@@ -0,0 +1,56 @@
+import React from 'react'
```
Comment (davidebianchi) [inline src/utilsTests.js]: We should begin with license

**Review by davidebianchi — CHANGES_REQUESTED** [review]

File: src/logo.svg (hunk 13 above)
Comment (Luca-Ferrera) [reply src/logo.svg:3]: It's the React's logo

**Review by davidebianchi — APPROVED** [review]

**Pull request merged**